### DISCUSSION TITLE
Resources (2024)

### DISCUSSION THREAD

**Discussion by Alex-Toucan** [ATProductsLLC/atproducts.xyz#211] [Announcements]
# Resources 
Resources are posted here so that things could be done easier.

<a href="https://www.youtube.com/watch?v=2qlcY9LkFik">Software Planning and Technical Documentation</a>

**Comment by Alex-Toucan** [comment]
## [Google WebFonts Helper](https://gwfh.mranftl.com/fonts)
> Self-host your Google fonts without sending analytics data to Google.

## [The modern JavaScript tutorial](https://javascript.info)

**Comment by Alex-Toucan** [comment]
## <a href="https://www.dbdiagram.io/home">Code Like DB Visualization</a>
> @VenenJean uses it all the time, but **why** did @VenenJean forget to post it here??? 🫥 🤔

**Comment by Alex-Toucan** [comment]
## [How to FavIcon](https://evilmartians.com/chronicles/how-to-favicon-in-2021-six-files-that-fit-most-needs)

**Comment by Alex-Toucan** [comment]
## Installing .NET 8.0 SDK on Ubuntu Server 24.04 LTS
`wget -O - https://packages.microsoft.com/keys/microsoft.asc | gpg --dearmor | sudo tee /etc/apt/keyrings/microsoft.gpg`

Create a file -> `/etc/apt/sources.list.d/microsoft.sources`

Paste content:
```js
Types: deb
URIs: https://packages.microsoft.com/ubuntu/22.04/prod/
Suites: jammy
Components: main
Architectures: amd64
Signed-By: /etc/apt/keyrings/microsoft.gpg
```
`sudo apt update`

### Fixing `libicu` dependency package issue
Package is at this date & time not available for Ubuntu 24.04 LTS
```bash
echo "deb http://archive.ubuntu.com/ubuntu jammy main" | sudo tee /etc/apt/sources.list.d/libicu-jammy.list
```
Pinning to lower priority to avoid conflicts with with main 24.04 repositories.
```bash
sudo tee /etc/apt/preferences.d/libicu <<EOF
Package: libicu*
Pin: release n=jammy
Pin-Priority: 100
EOF
```
Update system & install .NET SDK
```bash
sudo apt update && sudo apt install dotnet-sdk-8.0
```

**Comment by Alex-Toucan** [comment]
<@905434578926313512> (pinging myself) I (im insane) added:
[Javascript Mastery - YT](https://www.youtube.com/@javascriptmastery) -> High quality tutorials
[Javascript Mastery - Website](https://www.jsmastery.pro/) -> Free master resources
to https://discord.com/channels/1095979581787361380/1095996349293539349/1161564731501465610

**Comment by Alex-Toucan** [comment]
## [How to install and manage supervisor on Ubuntu and Debian VPS](https://www.digitalocean.com/community/tutorials/how-to-install-and-manage-supervisor-on-ubuntu-and-debian-vps)
Using for service management that run on my Ubuntu Server.

**Comment by Alex-Toucan** [comment]
## Apply mutational testing!!!
[How to test tests in C#](https://www.youtube.com/watch?v=sGwfwtkaDfk)
[Stryker.Net - Getting Started](https://stryker-mutator.io/docs/stryker-net/getting-started/)

**Comment by Alex-Toucan** [comment]
## [So you think you know Git?](https://www.youtube.com/watch?v=aolI_Rz0ZqY)

<sub>yes i do @venenjean, yes.</sub>

**Comment by Alex-Toucan** [comment]
## [Programmer Quiz](https://www.sporcle.com/games/moogles/programlanguages)

**Comment by Alex-Toucan** [comment]
## [progle()](https://progle.net/)
If you would like to guess programming languages based on a [Wordle](<https://www.nytimes.com/games/wordle/index.html>) type game, this is it for you. There's also a code color guessing game, which is based off the color code that programming languages have in certain text editors. This is really fun to play, so I suggest you try it out!

**Comment by Alex-Toucan** [comment]
[ The pro tutorials you’ve never seen if you use Blender ](https://www.youtube.com/watch?v=IHy57Lrzkb4)

@venenjean makes a total come back!!!

**Comment by Alex-Toucan** [comment]
## Cursor AI
[Software Developers are officially cooked](https://youtu.be/3wsVLG-dxg8?si=tKSH_9fla-15defj)

**Comment by Alex-Toucan** [comment]
## Some good stuff to learn code structure / logic / intention - whatever 
[SourceMaking.com](https://sourcemaking.com/)
[RefactoringGuru](https://refactoring.guru/)

**Comment by Alex-Toucan** [comment]
## [Making production ready C# applications](https://scottkuhl.medium.com/building-production-ready-applications-24405c706707)

**Comment by Alex-Toucan** [comment]
Why don't you become a contributor @venenjean?

**Reply by ghost** [reply]
Because - I planned out to move the resources someday to an archive website. So.... it's not worth to maintain 3 entities i.e. DC Channel, GitHub & (in the future) the website. (+ this one's your own issue xD)

**Reply by Alex-Toucan** [reply]
fair enough @VenenJean

**Comment by Alex-Toucan** [comment]
## [Business Name Generator](https://namelix.com/)

## [WhatTheFont](https://www.myfonts.com/)

## [BrandMark.Io](https://app.brandmark.io/v3/)

**Comment by Alex-Toucan** [comment]
## [Godot Docs](https://docs.godotengine.org/)

## [GitHub](https://github.com/)

## [GitLab](https://gitlab.com/)

## [GitKraken](https://www.gitkraken.com/)

## [BitBucket](https://bitbucket.org/)

## [Netlify](https://www.netlify.com/)

## [Vercel](https://vercel.com/)

## [Figma](https://figma.com/)

**Comment by Alex-Toucan** [comment]
## [PhotoPea](https://photopea.com)
Photoshop, but for free and online

**Comment by Alex-Toucan** [comment]
@venenjean just might aswell make a new godot resources accordion

## Godot UI setup [Godot, but prettier](https://www.youtube.com/watch?v=WmZq3UgOGKY)
Also the material theme from https://discord.com/channels/1095979581787361380/1291268878307164181/1291288320944504875
-> Consider his UI layout
## [A "toolbox" for Godot game dev](https://www.youtube.com/watch?v=rS88FPBtqBk)
## In Summary:

The full "stack" of components in a 3D project can be visualized as:

- **Model/Mesh:** Structure of the object.
- **Material/Shader/Texture:** How the object looks and interacts with light.
- **Rig/Animation:** How the object moves.
- **Scene/Light/Camera:** How the object is viewed and interacts in the environment.
- **Physics/Collider:** How the object interacts physically with other objects.
- **Scripts/Assets/Particles/Sound:** Control behaviors, effects, and environmental immersion.
## Terrain Rendering in Godot
> You know, I'm sorry, okay? But I'm learning Godot 3D Open World Game Dev and need to store resources somewhere.

[Mohsen Zare - MTerrain](https://www.youtube.com/watch?v=PcAkWClET4U)
[Mohsen Zare - Terrain Approach](https://www.youtube.com/watch?v=9olfCkPs1WU)
[Jen Jell - OWG](https://www.youtube.com/watch?v=MYINKjJV_G0&t=403s)
[Wandering Clipmap with LOD](https://www.youtube.com/watch?v=rcsIMlet7Fw)
[Terrain3D](https://www.youtube.com/watch?v=oV8c9alXVwU&t=551s)
[Wojtek - Forest Scene](https://www.youtube.com/watch?v=1ho6tbxGt4c&t=28s)

-> Getting strange errors with 1st and 4th...

**Comment by Alex-Toucan** [comment]
## BEST RESOURCE EVER!!! -> For beginners?
https://books.goalkicker.com/
|| Forgot to post it somehow... 😅 || @everyone

a bit... too much @VenenJean?

**Reply by Alex-Toucan** [reply]
also @everyone is a organization here on github lol

**Comment by Alex-Toucan** [comment]
## Game Shader
https://www.youtube.com/watch?v=UfSw6428bcc
[Using AI to Generate Game Assets For Free](https://www.youtube.com/watch?v=zEfBPmPDUVc)
## Bing AI on: What has to be on a model design sheet, so I can work with it in Blender?
```
Creating a model design sheet for use in Blender involves including several key elements to ensure you have all the necessary details for accurate 3D modeling. Here are the essential components:

1. Orthographic Views:
    Front View: Shows the model from the front.
    Side View: Displays the model from the side (left or right).
    Top View: Provides a view from above.
    Back View: Optional but useful for complex models.

2. Perspective View:
    A 3D perspective view to give a sense of depth and volume.

3. Proportions and Measurements:
    Include dimensions and scale references to ensure accuracy.

4. Details and Annotations:
    Highlight important features, textures, and materials.
    Annotate any specific details that need special attention.

5. Color and Texture References:
    Provide color palettes and texture samples that will be used in the final model.

6. Turnaround Sheets:
    For characters or complex objects, include turnaround sheets showing the model from multiple angles.

7. Pose Sheets (for characters):
    Include different poses to understand the range of motion and flexibility.

8. Additional Notes:
    Any additional information that might be useful during the modeling process.
```
## [Visual Library](https://characterdesignreferences.com/visual-library)
-> All about characters / anatomy - help for model design sheets

**Comment by Alex-Toucan** [comment]
## Free Assets
[Kenney](https://kenney.nl/)

**Comment by Alex-Toucan** [comment]
https://it-tools.tech/regex-memo

**Comment by Alex-Toucan** [comment]
## Button Types (Main site is about material design)
https://m3.material.io/components/all-buttons

**Comment by Alex-Toucan** [comment]
# RFC Editor
## Standards about the internet
https://www.rfc-editor.org/

A joke proposal, pigeons >> https://www.rfc-editor.org/rfc/rfc2549

**Comment by Alex-Toucan** [comment]
## [A helpful grouped Git Cheat Sheet](https://about.gitlab.com/images/press/git-cheat-sheet.pdf)

**Comment by Alex-Toucan** [comment]
## Found some free coding handbooks 🤷‍♂️ 
https://flaviocopes.com/access/

**Comment by Alex-Toucan** [comment]
## [Web Formatter](https://webformatter.com/)
A pretty good one, if you ask me. 😉

**Comment by Alex-Toucan** [comment]
Reminder this is moved to #385
